### PR TITLE
Add Collapse component to Palette

### PR DESCRIPTION
**Pull request by sepans**
Moves `Collapse` component from Reaction to Palette to be used by **volt**.

This is exactly the copy paste of the Reaction [implementation](https://github.com/artsy/reaction/blob/master/src/Styleguide/Components/Collapse.tsx).

@yuki24 brought up a point that the palette repo is meant to be very light in terms of dependency but `react-spring` that is used for collapsing is a relatively large one so:

- If this kind of animations will be used in future palette components we can keep `react-spring` here
- Or use an alternative library or implementation
- Or keep `Collapse` in reaction and import it from there into volt, although reaction is becoming a very heavy dependency itself, to be imported as a component library.

screen shot 2018-11-08 at 4 29 53 pm

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,6 +87,7 @@
   "dependencies": {
     "rc-slider": "^8.6.2",
     "react": "^16.5.0",
+    "react-spring": "^5.7.2",
```
Comment (damassi) [inline package.json]: Palette will have an animation section fairly shortly so it was expected that this dependency would be added in time.

File: src/elements/Collapse/Collapse.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React from "react"
+import { animated, Spring } from "react-spring"
+
+export interface CollapseProps {
+  /** is open? */
```
Comment (damassi) [inline src/elements/Collapse/Collapse.tsx]: `/** Determines whether content is expanded or collapsed /*`

**Comment by artsyit** [comment]
Deploy preview for *artsy-palette* ready!

Built with commit 4ed412bc8641363431b98152bfbcc7d4fe1eea01

https://deploy-preview-126--artsy-palette.netlify.com

File: src/elements/Collapse/Collapse.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  open: boolean
+}
+
+/** Collapse component */
```
Comment (damassi) [inline src/elements/Collapse/Collapse.tsx]: This comes through in VSCode tooling and eventually docs; mind expanding this decription a bit?

File: docs/Collapse.mdx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+# Collapse
+
+## Open
+
+<Playground>
+    <Collapse open>
+      <BorderBox>
+        A Collapse component ...
+        Not really cool yet!
```
Comment (damassi) [inline docs/Collapse.mdx]: It's cool!

**Comment by damassi** [comment]
Ahh @sepans - message is case sensitive: 

`[FEATURE Collapse] Add Collapse component to Palette`

**Pull request merged**